### PR TITLE
opx-sai-vm (17.2.0)

### PR DESCRIPTION
**Pull request by rakeshdatta**
* Update: 1d Bridge RIF and 1D router bridge port support
  * Update: Ported Latest Trap/User trap sai-bcm code into sai-vm repo
  * Update: Added Bridge type qualifier into SAI attribute list

Signed-off-by: Rakesh Datta <rakesh.datta@dell.com>

### CODE REVIEW

File: opx/sai_ipmc_common.h [hunk 1]
```diff
@@ -0,0 +1,172 @@
+/************************************************************************
+* * LEGALESE:   "Copyright (c) 2017, Dell Inc. All rights reserved."
```
Comment (atanu-mandal) [inline opx/sai_ipmc_common.h]: Can you update the year of Copyright.

File: opx/sai_l3_mcast_common.h [hunk 2]
```diff
@@ -0,0 +1,115 @@
+/************************************************************************
+* * LEGALESE:   "Copyright (c) 2017, Dell Inc. All rights reserved."
```
Comment (atanu-mandal) [inline opx/sai_l3_mcast_common.h]: Can you update the year of Copyright.

File: opx/sai_npu_ipmc.h [hunk 3]
```diff
@@ -0,0 +1,141 @@
+/************************************************************************
+* * LEGALESE:   "Copyright (c) 2017, Dell Inc. All rights reserved."
```
Comment (atanu-mandal) [inline opx/sai_npu_ipmc.h]: Can you update the year of Copyright.

File: opx/sai_npu_l3_mcast.h [hunk 4]
```diff
@@ -0,0 +1,79 @@
+/************************************************************************
+* * LEGALESE:   "Copyright (c) 2017, Dell Inc. All rights reserved."
```
Comment (atanu-mandal) [inline opx/sai_npu_l3_mcast.h]: Can you update the year of Copyright.

File: src/routing/sai_l3_ipmc.c [hunk 5]
```diff
@@ -0,0 +1,118 @@
+/************************************************************************
+* * LEGALESE:   "Copyright (c) 2017, Dell Inc. All rights reserved."
```
Comment (atanu-mandal) [inline src/routing/sai_l3_ipmc.c]: Can you update the year of Copyright.

File: src/routing/sai_l3_ipmc_group.c [hunk 6]
```diff
@@ -0,0 +1,193 @@
+/************************************************************************
+* * LEGALESE:   "Copyright (c) 2017, Dell Inc. All rights reserved."
```
Comment (atanu-mandal) [inline src/routing/sai_l3_ipmc_group.c]: Please update the year.

File: src/routing/sai_l3_ipmc_repl_group.c [hunk 7]
```diff
@@ -0,0 +1,114 @@
+/************************************************************************
+* * LEGALESE:   "Copyright (c) 2017, Dell Inc. All rights reserved."
```
Comment (atanu-mandal) [inline src/routing/sai_l3_ipmc_repl_group.c]: Can you update the year of Copyright.

File: src/routing/sai_l3_ipmc_rpf_group.c [hunk 8]
```diff
@@ -0,0 +1,185 @@
+/************************************************************************
+* * LEGALESE:   "Copyright (c) 2017, Dell Inc. All rights reserved."
```
Comment (atanu-mandal) [inline src/routing/sai_l3_ipmc_rpf_group.c]: Can you update the year of Copyright.

File: src/routing/sai_vm_l3_ipmc.c [hunk 9]
```diff
@@ -0,0 +1,120 @@
+/************************************************************************
+* * LEGALESE:   "Copyright (c) 2017, Dell Inc. All rights reserved."
```
Comment (atanu-mandal) [inline src/routing/sai_vm_l3_ipmc.c]: Can you update the year of Copyright.

**Comment by rakeshdatta** [comment]
Pull-request updated, HEAD is now fd0e71b313ca13f591c56148b4b6538019543db8

File: opx/sai_l3_mcast_common.h (hunk 2 above)
Comment (rakeshdatta) [reply opx/sai_l3_mcast_common.h]: Fixed

**Review by atanu-mandal — APPROVED** [review]

**Pull request merged**